pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
I don't get how that follows. Can't you just as well type assert to a `Validator` for that purpose without having to pollute these interface definitions? I would much rather have interface definitions be as narrow as possible.